src/main/java/io/toadlabs/numeralping/mixin/ServerEntryMixin.java

```diff
@@ -35,12 +34,9 @@ public void shiftText(Args args) {
 	}
 
 	// hide the tooltip if it's redundant
-	@ModifyArgs(method = "render", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/multiplayer/MultiplayerScreen;setMultiplayerScreenTooltip(Ljava/util/List;)V"), require = 0)
-	public void setTooltip(Args args) {
-		var tooltip = (List<Text>) args.get(0);
-		if (tooltip != null && tooltip.size() == 1 && NumeralConfig.instance().serverList && tooltip.get(0).getContent() instanceof TranslatableTextContent translatable && translatable.getKey().equals("multiplayer.status.ping")) {
-			args.set(0, null);
-		}
+	@WrapWithCondition(method = "render", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/multiplayer/MultiplayerScreen;setTooltip(Lnet/minecraft/text/Text;)V"))
+	public boolean hideTooltip(MultiplayerScreen instance, Text text) {
+		return !NumeralConfig.instance().serverList;
```

Comment:
The `multiplayer.status.ping` was there for a good reason - only the ping is redundant but other information can be displayed

Reply:
i foror